beacon_node/beacon_chain/src/data_availability_checker/processing_cache.rs

```diff
@@ -0,0 +1,92 @@
+use crate::data_availability_checker::AvailabilityView;
+use ssz_types::FixedVector;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::sync::Arc;
+use types::beacon_block_body::{KzgCommitmentOpts, KzgCommitments};
+use types::blob_sidecar::{BlobIdentifier, FixedBlobSidecarList};
+use types::signed_beacon_block::get_missing_blob_ids;
+use types::{EthSpec, Hash256, SignedBeaconBlock, Slot};
+
+/// This cache is used only for gossip blocks/blobs and single block/blob lookups, to give req/resp
+/// a view of what we have and what we require. This cache serves a slightly different purpose than
+/// gossip caches because it allows us to process duplicate blobs that are valid in gossip.
+/// See `AvailabilityView`'s trait definition.
+#[derive(Default)]
+pub struct ProcessingCache<E: EthSpec> {
+    processing_cache: HashMap<Hash256, ProcessingInfo<E>>,
+}
+
+impl<E: EthSpec> ProcessingCache<E> {
+    pub fn get(&self, block_root: &Hash256) -> Option<&ProcessingInfo<E>> {
+        self.processing_cache.get(block_root)
+    }
+    pub fn entry(&mut self, block_root: Hash256) -> Entry<'_, Hash256, ProcessingInfo<E>> {
+        self.processing_cache.entry(block_root)
+    }
+    pub fn remove(&mut self, block_root: &Hash256) {
+        self.processing_cache.remove(block_root);
+    }
+    pub fn has_block(&self, block_root: &Hash256) -> bool {
+        self.processing_cache
+            .get(block_root)
+            .map_or(false, |b| b.kzg_commitments.is_some())
```

Comment:
u might consider changing this to `b.block_exists()`
```suggestion
            .map_or(false, |b| b.block_exists())
```